### ISSUE TITLE
Support keyboard navigation

### ISSUE THREAD

**Issue by localpcguy** [cibernox/ember-basic-dropdown#212]
While I recognize this has a broader scope than just as a form element replacement, it will likely be used for that purpose pretty often. Given that, most standard browser dropdowns allow for keyboard navigation, which helps with accessibility

- up/down arrows open the dropdown and position the focus on the last/first item respectively
- up/down arrows navigate up and down the dropdown items after the dropdown is open
- typing any letter should focus the first element whose first letter matches
- typing any letter again should focus the next element whose first letter matches
- hitting space or enter when an item is focused selects that item and closes the dropdown
- hitting tab when an item is focused selects that item, closes the dropdown and moves to the next element in the page's taborder

**Comment by cibernox** [comment]
I think this issue was meant to another project. The dropdown has no such thing as "items", so nothing of the above makes any sense.
If you are looking for a select check www.ember-power-select.com

**Comment by localpcguy** [comment]
Each "thing" in the dropdown is an item. ember-power-select is likely a better choice for selects, but I guarantee people will use this to customize them as well. Regardless, thanks for creating open source code and releasing it!

**Comment by cibernox** [comment]
@localpcguy Yes, I can imagine people will use the dropdown to create things like menus. At the end I extracted this addon when I built the select component. But dropdowns can also contain calendars, just plain text and a photo... 
It can really contain anything, so it's beyond the scope of this component to worry about items and arrow keys. That is functionality users will have to build on top if they want, all the necessary hooks are present.
